spock-core/src/main/java/org/spockframework/compiler/WhereBlockRewriter.java

```diff
@@ -413,16 +534,42 @@ private void rewriteExpressionTableLikeParameterization(ListIterator<Statement>
   private void rewriteTableLikeParameterization(ListIterator<Statement> stats, Function<List<Expression>, Boolean> rowExtractor) throws InvalidSpecCompileException {
     LinkedList<List<Expression>> rows = new LinkedList<>();
 
+    // iterate rows until where block ends
     while (stats.hasNext()) {
+      if (rows.size() > 0) {
+        // or the next statement is combined-labeled after the table has started already, i.e. a new data table
+        // or data pipe starts that should be cross-multiplied so the current table is finished
+        boolean combinedLabeled = COMBINED.equals(stats.next().getStatementLabel());
```

Comment:
How about a method `isCombinedStatement(Statement)`
The same question is asked at least 3 times.